spec/index.bs

```diff
@@ -6071,36 +6075,48 @@ interface mixin GPUPipelineBase {
+                **Called on:** {{GPUPipelineBase}} |this|
 
-            **Arguments:**
+                **Arguments:**
 
-            <pre class=argumentdef for="GPUPipelineBase/getBindGroupLayout(index)">
-                |index|: Index into the pipeline layout's {{GPUPipelineLayout/[[bindGroupLayouts]]}}
-                    sequence.
-            </pre>
+                <pre class=argumentdef for="GPUPipelineBase/getBindGroupLayout(index)">
+                    |index|: Index into the pipeline layout's {{GPUPipelineLayout/[[bindGroupLayouts]]}}
+                        sequence.
+                </pre>
 
-            **Returns:** {{GPUBindGroupLayout}}
+                **Returns:** {{GPUBindGroupLayout}}
 
-            1. If |index| &ge;
-                |this|.{{GPUObjectBase/[[device]]}}.{{device/[[limits]]}}.{{supported limits/maxBindGroups}}:
-                1. Throw a {{RangeError}}.
+                [=Content timeline=] steps:
 
-            1. If |this| is not [=valid=]:
-                1. Return a new error {{GPUBindGroupLayout}}.
+                1. If |index| &ge;
+                    |this|.{{GPUObjectBase/[[device]]}}.{{device/[[limits]]}}.{{supported limits/maxBindGroups}}:
+                    1. Throw a {{RangeError}}.
+                1. Let |layout| be a new {{GPUBindGroupLayout}} object.
+                1. Issue the |initialization steps| on the [=Device timeline=] of |this|.
+                1. Return |layout|.
+            </div>
+            <div data-timeline=device>
+                [=Device timeline=] |initialization steps|:
 
-            1. If |index| &ge; the [=list/size=] of
-                |this|.{{GPUPipelineBase/[[layout]]}}.{{GPUPipelineLayout/[[bindGroupLayouts]]}}:
-                1. Return a new error {{GPUBindGroupLayout}}.
+                1. If any of the following conditions are unsatisfied
+                    [$generate a validation error$], make |layout| [=invalid=], and stop.
+                
+                    <div class=validusage>
+                        - |this| is [=valid=].
+                        - |index| &lt; the [=list/size=] of
+                            |this|.{{GPUPipelineBase/[[layout]]}}.{{GPUPipelineLayout/[[bindGroupLayouts]]}}
+                    </div>
 
-            1. Return a new {{GPUBindGroupLayout}} object that references the same internal object as
-                |this|.{{GPUPipelineBase/[[layout]]}}.{{GPUPipelineLayout/[[bindGroupLayouts]]}}[|index|].
+                1. Update |layout| to reference the same internal object as
+                    |this|.{{GPUPipelineBase/[[layout]]}}.{{GPUPipelineLayout/[[bindGroupLayouts]]}}[|index|].
 
-            Issue: Specify this more properly once we have internal objects for {{GPUBindGroupLayout}}.
-                Alternatively only spec is as a new internal objects that's [=group-equivalent=]
+                Issue: Specify this more properly once we have internal objects for {{GPUBindGroupLayout}}.
```

Comment:
aside: The way I've been thinking about defining the server->client references wouldn't work with this because this requires making multiple client references point to the same server object - can't just "allocate" an object from the client and then "initialize" it from the server.

However I'm pretty sure this is not actually a problem. BGL objects don't have to be reference-equal, they only have to be value-equal ("[group-equivalent](https://gpuweb.github.io/gpuweb/#group-equivalent)").